src/Nextjs.ts

```diff
@@ -156,7 +156,7 @@ export class Nextjs extends Construct {
     });
     // finish static deployment BEFORE deploying new function code
     // as there is some time after the new static files are uploaded but before they are rewritten
-    this.assetsDeployment.node.addDependency(this.serverFunction);
+    this.serverFunction.node.addDependency(this.assetsDeployment);
```

Comment:
The assets might be uploaded to S3 but not rewritten yet, I'm trying to see if it's possible to have a dependency on the rewriter but it's a little tricky

Reply:
Seems like maybe it should be a dependency on the rewriter if we're doing rewrites (env vars provided) or just the bucket deployment if not according to someone